src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1500,6 +1501,8 @@ public void updateLatestADTask(
         getAndExecuteOnLatestDetectorLevelTask(detectorId, taskTypes, (adTask) -> {
             if (adTask.isPresent()) {
                 updateADTask(adTask.get().getTaskId(), updatedFields);
+            } else {
```

Comment:
some generics issue for this method:
1) the parameter listener should have generic type.
2) getAndExecuteOnLatestDetectorLevelTask is a generic method.  Please add the type.